htdocs/admin/fichinter_xcal.php

```diff
@@ -0,0 +1,244 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ *	    \file       htdocs/admin/fichinter_xcal.php
+ *      \ingroup    agenda
+ *      \brief      Page to setup miscellaneous options of agenda module
+ */
+
+// Load Dolibarr environment
+require '../main.inc.php';
+require_once DOL_DOCUMENT_ROOT.'/core/lib/admin.lib.php';
+require_once DOL_DOCUMENT_ROOT.'/core/lib/fichinter.lib.php';
+
+
+if (!$user->admin) {
+	accessforbidden();
+}
+
+// Load translation files required by the page
+$langs->loadLangs(array("admin", "other", "agenda"));
+
+$def = array();
+$actionsave = GETPOST('save', 'alpha');
+$MAIN_FICHINTER_XCAL_EXPORTKEY = getDolGlobalString('MAIN_FICHINTER_XCAL_EXPORTKEY');
+$MAIN_FICHINTER_EXPORT_PAST_DELAY = getDolGlobalString('MAIN_FICHINTER_EXPORT_PAST_DELAY', 100);
+$MAIN_FICHINTER_EXPORT_CACHE = getDolGlobalInt('MAIN_FICHINTER_EXPORT_CACHE');
+$MAIN_FICHINTER_EXPORT_FIX_TZ = getDolGlobalString('MAIN_FICHINTER_EXPORT_FIX_TZ');
+
+if (GETPOSTISSET('MAIN_FICHINTER_XCAL_EXPORTKEY')) {
+	$MAIN_FICHINTER_XCAL_EXPORTKEY = trim(GETPOST('MAIN_FICHINTER_XCAL_EXPORTKEY', 'alpha'));
+}
+if (GETPOSTISSET('MAIN_FICHINTER_EXPORT_PAST_DELAY')) {
+	$MAIN_FICHINTER_EXPORT_PAST_DELAY = intval(GETPOSTINT('MAIN_FICHINTER_EXPORT_PAST_DELAY'));
+}
+if (GETPOSTISSET('MAIN_FICHINTER_EXPORT_CACHE')) {
+	$MAIN_FICHINTER_EXPORT_CACHE = intval(GETPOSTINT('MAIN_FICHINTER_EXPORT_CACHE'));
+}
+if (GETPOSTISSET('MAIN_FICHINTER_EXPORT_FIX_TZ')) {
+	$MAIN_FICHINTER_EXPORT_FIX_TZ = trim(GETPOST('MAIN_FICHINTER_EXPORT_FIX_TZ', 'alpha'));
+}
+
+// Sauvegardes parameters
+if ($actionsave) {
+	$i = 0;
+
+	$db->begin();
+
+	$i += dolibarr_set_const($db, 'MAIN_FICHINTER_XCAL_EXPORTKEY', $MAIN_FICHINTER_XCAL_EXPORTKEY, 'chaine', 0, '', $conf->entity);
+	$i += dolibarr_set_const($db, 'MAIN_FICHINTER_EXPORT_PAST_DELAY', $MAIN_FICHINTER_EXPORT_PAST_DELAY, 'chaine', 0, '', $conf->entity);
+	$i += dolibarr_set_const($db, 'MAIN_FICHINTER_EXPORT_CACHE', $MAIN_FICHINTER_EXPORT_CACHE, 'chaine', 0, '', $conf->entity);
+	$i += dolibarr_set_const($db, 'MAIN_FICHINTER_EXPORT_FIX_TZ', $MAIN_FICHINTER_EXPORT_FIX_TZ, 'chaine', 0, '', $conf->entity);
+
+	if ($i >= 4) {
+		$db->commit();
+		setEventMessages($langs->trans("SetupSaved"), null, 'mesgs');
+	} else {
+		$db->rollback();
+		setEventMessages($langs->trans("SaveFailed"), null, 'errors');
+	}
+}
+
+
+
+/**
+ * View
+ */
+
+if (!isset($conf->global->MAIN_FICHINTER_EXPORT_PAST_DELAY)) {
```

Comment:
Use conf->global is now forbiden. Can you replace with getDolGlobalString ?

Reply:
change made